app/javascript/controllers/dashboard_controller.js

```diff
@@ -0,0 +1,149 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class DashboardController extends Controller {
+  static targets = [
+    "form", "name", "category", "price", "vegetarian", "image",
+    "clearButton", "resetButton", "tableBody", "submitButton"
+  ]
+
+  connect() {
+    console.log("✅ Stimulus dashboard controller connected");
+    this.foodItems = [];
+    this.editingItemId = null;
+    this.initDashboard();
+  }
+
+  async initDashboard() {
+    try {
+      let resp = await fetch("/food_items", { credentials: "same-origin" });
+      if (!resp.ok) throw new Error(await resp.text());
+      this.foodItems = await resp.json();
+    } catch (e) {
+      console.error("Couldn't load food items:", e);
+      this.foodItems = [];
+    }
+    this.renderFoodTable();
+  }
+
+  resetToBaseline() {
+    if (window.foodItems && Array.isArray(window.foodItems)) {
+      this.foodItems = JSON.parse(JSON.stringify(window.foodItems));
+    } else {
+      this.foodItems = [];
+    }
+    this.saveFoodItems();
+    this.renderFoodTable();
+  }
+
+  saveFoodItems() {
+    localStorage.setItem(this.FOOD_ITEMS_STORAGE_KEY, JSON.stringify(this.foodItems));
+  }
+
+  renderFoodTable() {
+    this.tableBodyTarget.innerHTML = '';
+
+    if (this.foodItems.length === 0) return;
+
+    this.foodItems.forEach(item => {
+      const row = document.createElement('tr');
+      row.innerHTML = `
```

Comment:
I wouldn't say populating the views from Stimulus this way is the best approach in this case. I think rendering them usin a .erb file is the way to go